README.md

```diff
@@ -27,8 +37,290 @@ pip install -e .
+```
+
+<p align="center">
+<img src="Figures/cleaning_params.png" width="50%"/>
+</p>
+
+If the cleaning parameters seem well located (cyan cross on the centre, sky
+radius outside the fringes pattern, etc.), we can apply the cleaning step to the
+data.
+
+```python
+
+cube_cleaned = amical.select_clean_data(nrm_file, **clean_param, clip=True, clip_fact=0.5)
+```
+
+During the cleaning step, you can decide to apply a lucky imaging selection
+(`clip`=True) to accumulate only the best frames in the cube (based on the
+integrated fluxes compared to the median: threshold = median(fluxes) -
+`clip_fact` x std(fluxes)).
+
+<p align="center">
+<img src="Figures/clipping.png" width="80%"/>
+</p>
+
+### Step 2: extract observables
+
+The second step is the core of AMICAL: we use the Fourier sampling approach to
+extract the interferometric observables (visibilities and closure phases).
+
+All the challenge when you play with the NRM data is to find the correct
+position of each baseline in the Fourier transform. To do so, we implemented 4
+different sampling methods (`peakmethod` = ('unique', 'square', 'gauss', 'fft'))
+to exploit information spread beyond just the _u_, _v_ positions.
+
+<p align="center">
+<img src="Figures/fft.png" width="60%"/>
+</p>
+<p align="center">
+<img src="Figures/peakmethod.png" width="80%"/>
+</p>
+
+Based on NIRISS and SPHERE dataset analysis, we recommend using the **fft**
+method (but feel free to test the other methods for your specific case!). The
+expected baseline locations on the detector are computed using the mask
+coordinates, the wavelength and the pixel size. In some cases, the mask is not
+perfectly aligned with the detector and so requires to be rotated
+(`theta_detector` = 0) or centrally scaled (`scaling_uv` = 1). 
+
+With AMICAL, the mask coordinates, the wavelengths, the pixel size and the
+target name are normally determined using the fits header informations.
+Otherwise, you will need to give `filtname`,`instrum` and `targetname` to
+determine those information from the AMICAL internal database
+([get_infos_obs.py](amical/get_infos_obs.py)).
+
+```python
+params_ami = {"peakmethod": "fft",
+              "maskname": "g7", # 7 holes mask of NIRISS
+              }
+
+bs = amical.extract_bs(cube_cleaned, file_t, **params_ami)
+```
+
+> Note: Other parameters of `amical.extract_bs()` are rarely modified but you
+> can check the docstrings for details
+> ([bispect.py](amical/mf_pipeline/bispect.py)).
+
+The object `bs` stores the raw observables (`bs.vis2`, `bs.e_vis2`, `bs.cp`,
+`bs.e_cp`), the u-v coordinates and wavelength (`bs.u`, `bs.v`, `bs.wl`), the
+baseline lengths (`bs.bl`, `bs.bl_cp`), information relative to the used mask
+(`bs.mask`), the computed matrices and statistic (`bs.matrix`) and the important
+information (`bs.infos`). The .mask, .infos and .matrix are also class with
+various quantities.
+
+```python
+print bs.keys(), bs.mask.keys() # for details
```

Comment:
Corrected for python 3...
```suggestion
print(bs.keys(), bs.mask.keys())
```
note that if `bs` and `bs.mask` behave as a builtin dicts, I expect this to print something like
```
dict_keys([...]), dict_keys([...])
```
which is pretty ugly